components/match2/commons/starcraft_starcraft2/match_group_input_starcraft.lua

```diff
@@ -371,81 +370,59 @@ end
 ---@param mapInput table
 ---@param opponent table
 ---@param opponentIndex integer
----@return table<string, {faction: string?, player: string, position: string, flag: string?}>
+---@return table<string, {faction: string?, player: string, flag: string?}>
 function MapFunctions.getTeamParticipants(mapInput, opponent, opponentIndex)
-	local players = opponent.match2players
-
 	local archonFaction = Faction.read(mapInput['t' .. opponentIndex .. 'p1race'])
 		or Faction.read(mapInput['opponent' .. opponentIndex .. 'race'])
-		or ((players[1] or {}).extradata or {}).faction
+		or ((opponent.match2players[1] or {}).extradata or {}).faction
 	local isArchon = MapFunctions.isArchon(mapInput, opponent, opponentIndex)
 
-	---@type {input: string, faction: string?, link: string?}[]
-	local participantsList = Array.mapIndexes(function(playerIndex)
+	local players = Array.mapIndexes(function(playerIndex)
+		if opponent.match2players[playerIndex] then
+			return opponent.match2players[playerIndex]
+		end
 		local prefix = 't' .. opponentIndex .. 'p' .. playerIndex
 
 		if Logic.isEmpty(mapInput[prefix]) then return end
 
 		return {
-			input = mapInput[prefix],
-			link = Logic.nilIfEmpty(mapInput[prefix .. 'link']),
+			displayname = mapInput[prefix],
+			name = Logic.nilIfEmpty(mapInput[prefix .. 'link']) or Variables.varDefault(mapInput[prefix] .. '_page'),
 			faction = isArchon and archonFaction or Faction.read(mapInput[prefix .. 'race']),
 		}
-	end)
-
-	local participants = {}
-
-	Array.forEach(participantsList, function(participantInput, position)
-		local nameInput = participantInput.input
-
-		local isTBD = nameInput:upper() == TBD or nameInput:upper() == TBA
-
-		local link = participantInput.link or Variables.varDefault(nameInput .. '_page') or nameInput
-		link = Page.pageifyLink(link) --[[@as string -- can't be nil as input isn't nil]]
-
-		local playerIndex = MapFunctions.getPlayerIndex(players, link, nameInput)
-
-		-- in case we have a TBD or a player not known in match2players inster a new player in match2players
-		if isTBD or playerIndex == 0 then
-			table.insert(players, {
-				name = isTBD and TBD or link,
-				displayname = isTBD and TBD or nameInput,
-				extradata = {faction = participantInput.faction or Faction.defaultFaction},
-			})
-			playerIndex = #players
-		end
+	end) --[[@as table[]]
 
+	local participants, unattachedParticipants = MatchGroupInputUtil.parseParticipants(players, function(playerIndex)
 		local player = players[playerIndex]
 
-		participants[opponentIndex .. '_' .. playerIndex] = {
-			faction = participantInput.faction or player.extradata.faction,
-			player = link,
-			position = position,
+		return {
+			faction = player.faction or player.extradata.faction,
```

Comment:
this doesn't make much sense
fwiw it is not doable in this seup because you need the match2players[playerId] data here too